File: ComponentDemo/src/oracle/adfdemo/view/explorer/rich/validator/ShuttleValidator.java
/* Copyright (c) 2008, 2014, Oracle and/or its affiliates. 
All rights reserved.*/

package oracle.adfdemo.view.explorer.rich.validator;

import java.awt.List;

import java.util.ArrayList;
import java.util.Collection;

import javax.faces.application.FacesMessage;
import javax.faces.component.UIComponent;
import javax.faces.context.FacesContext;
import javax.faces.validator.Validator;
import javax.faces.validator.ValidatorException;

import org.apache.myfaces.trinidad.validator.ClientValidator;

public class ShuttleValidator implements Validator, ClientValidator
{
  public void validate(FacesContext context, UIComponent component, Object value) throws ValidatorException
  {
    ArrayList topActors = (ArrayList) value;
    
    if (topActors.size() > 5)
      throw new ValidatorException(new FacesMessage("Please limit your selection to 5 actors!", ""));
  }

  public Collection<String> getClientImportNames()
  {
    return null;
  }

  public String getClientLibrarySource(FacesContext context)
  {
    return context.getExternalContext().getRequestContextPath() + 
    "/fileExplorer/jsLibs/shuttleValidator.js";
  }

  public String getClientScript(FacesContext context, UIComponent component)
  {
    return null;
  }

  public String getClientValidation(FacesContext context, UIComponent component)
  {
    return ("new ShuttleValidator()");
  }

}

File: ComponentDemo/public_html/fileExplorer/jsLibs/shuttleValidator.js
/*
** Copyright (c) 2008, 2014, Oracle and/or its affiliates. All rights reserved.
*/
function ShuttleValidator()
{
  // for debugging
  this._class = "ShuttleValidator";
}

ShuttleValidator.prototype = new TrValidator();

ShuttleValidator.prototype.validate = function(value, label, converter)
{
  var nActors = value.size();
   
  if (size > 5)
  {
    var fm = new TrFacesMessage("Please limit your selection to 5 file types!", 
                                "client", 
                                TrFacesMessage.SEVERITY_ERROR);
    throw new TrValidatorException(fm);
  }
}